packages/frontend-2/components/settings/workspaces/automation/Functions/TableRowActions.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <LayoutMenu
+    v-model:open="isOpen"
+    :items="actionItems"
+    mount-menu-on-body
+    :menu-position="HorizontalDirection.Left"
+    @chosen="({ item }) => handleAction(item)"
+  >
+    <FormButton
+      :color="isOpen ? 'outline' : 'subtle'"
+      hide-text
+      :icon-right="isOpen ? XMarkIcon : EllipsisHorizontalIcon"
+      @click.stop="isOpen = true"
+    />
+    <SettingsWorkspacesAutomationFunctionsRegenerateTokenDialog
+      v-model:open="showRegenerateTokenDialog"
+      :workspace-function="workspaceFunction"
+    />
+  </LayoutMenu>
+</template>
+
+<script setup lang="ts">
+import type { LayoutMenuItem } from '@speckle/ui-components'
+import { HorizontalDirection } from '@speckle/ui-components'
+import { EllipsisHorizontalIcon, XMarkIcon } from '@heroicons/vue/24/outline'
+import type { SettingsWorkspacesAutomationTableRowActions_AutomateFunctionFragment } from '~/lib/common/generated/gql/graphql'
+import { graphql } from '~/lib/common/generated/gql'
+
+graphql(`
+  fragment SettingsWorkspacesAutomationTableRowActions_AutomateFunction on AutomateFunction {
+    id
+    permissions {
+      ...SettingsWorkspacesAutomationTableRowActions_AutomateFunctionPermissionChecks
+    }
+    ...SettingsWorkspacesAutomationRegenerateTokenDialog_AutomateFunction
+  }
+`)
+
+graphql(`
+  fragment SettingsWorkspacesAutomationTableRowActions_AutomateFunctionPermissionChecks on AutomateFunctionPermissionChecks {
+    canRegenerateToken {
+      authorized
```

Comment:
not a big deal, but there's a `...FullPermissionCheckResult` fragment that you can use for auth checks